app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/AddProductNutritionFactsFragment.java

```diff
@@ -1070,6 +1071,7 @@ private void addNutrientRow(int position, CharSequence text, boolean preFillValu
         editText.setHint(text);
         editText.setId(position);
         editText.setKeyListener(keyListener);
+        editText.setImeOptions(EditorInfo.IME_ACTION_DONE);
```

Comment:
I guess we won't be able to move to the next nutrition facts field using the soft key if we add this to each extra nutrient we add.

Reply:
@huzaifaiftikhar If I've selected multiple nutrients and put IME_ACTION-DONE in addNutrientRow() method then, all the added nutrient will dismiss keyboard when press IME button. The only way I think of it is to store all EditText in an ArrayList and then put a check that, if the position is n-1 then put IME_ACTION_DONE else nextFocusDown. But the risk is that, if the ArrayList is empty then it will crash the application. So, guide me on this issue.

Reply:
@jaindiv26 I guess if we add `IME_ACTION_NEXT` instead it will work too. After the last field the focus goes back to the first field. Also I am not sure if this would solve the issue which this PR is targeting. Even in the absence of ime options the app should not crash. What's your thought on this?

Reply:
@huzaifaiftikhar IME.ACTION.DONE or NEXT calls to find the next focus item that should be quickly jumped to, but this does not exist or isn't focusable, it throws this error (java.lang.IllegalStateException android.widget.TextView.onKeyUp). It will happen if the next is hidden and thus not focusable or the parent cannot actually deal with the findFocus() call and returns null.

https://stackoverflow.com/questions/13614101/fatal-crash-focus-search-returned-a-view-that-wasnt-able-to-take-focus/14189135

Reply:
Thanks for the link. I read all the answers however I still have a doubt. This is happening when the user has defined the ime options either to NEXT or PREVIOUS, however in our case we have not defined the ime options and it has a default behaviour. According to your answer it means that it is necessary to define the ime options to every edit text we use in an android app or else the app would crash. This doesn't seem correct to me. Do you have a device where you can actually reproduce this?

Reply:
@huzaifaiftikhar I've tried every scenario on various devices but, can't reproduce this issue, I think this issue happens on devices with a hardware keyboard support. I've one possible solution in mind, let me implement it and see the results.

Reply:
@jaindiv26 It seems like the issue is not limited to devices with a hardware keyboard. We have crash reports for devices with a soft keyboard as well. Check the latest comments on the [issue](https://github.com/openfoodfacts/openfoodfacts-androidapp/issues/1826).

Reply:
@huzaifaiftikhar @Karljoones Let me try it, one more time.